development/tools/pkg/jobguard/status.go

```diff
@@ -0,0 +1,107 @@
+package jobguard
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"log"
+	"net/http"
+
+	"github.com/pkg/errors"
+)
+
+// StatusState is state of a status (job)
+type StatusState string
+
+// StatusStateSuccess represents success state of a status
+const StatusStateSuccess StatusState = "success"
+
+// StatusStatePending represents pending state of a status
+const StatusStatePending StatusState = "pending"
+
+// StatusStateError represents error state of a status
+const StatusStateError StatusState = "error"
+
+// StatusStateFailure represents failure state of a status
+const StatusStateFailure StatusState = "failure"
+
+// Status stores essential data for a status
+type Status struct {
+	Name  string `json:"context"`
+	State string `json:"state"`
+}
+
+// StatusConfig holds configuraiton for GithubStatusFetcher
+type StatusConfig struct {
+	Origin     string `envconfig:"default=https://api.github.com,API_ORIGIN"`
+	Owner      string `envconfig:"default=kyma-project,REPO_OWNER"`
+	Repository string `envconfig:"default=kyma,REPO_NAME"`
+	CommitSHA  string `envconfig:"COMMIT_SHA"`
+}
+
+// GithubStatusFetcher fetches all statuses for the given commit
+type GithubStatusFetcher struct {
+	cfg    StatusConfig
+	client *http.Client
+}
+
+// NewStatusFetcher constructs new GithubStatusFetcher instance
+func NewStatusFetcher(cfg StatusConfig, client *http.Client) *GithubStatusFetcher {
+	return &GithubStatusFetcher{cfg: cfg, client: client}
+}
+
+type statusResponse struct {
+	TotalCount int      `json:"total_count"`
+	Statuses   []Status `json:"statuses"`
+}
+
+// Do fetches Github statuses
+func (f *GithubStatusFetcher) Do() ([]Status, error) {
+	url := fmt.Sprintf("%s/repos/%s/%s/commits/%s/status", f.cfg.Origin, f.cfg.Owner, f.cfg.Repository, f.cfg.CommitSHA)
+
+	var statuses []Status
+	pageNo := 1
+
+	for {
+		pageURL := fmt.Sprintf("%s?page=%d&per_page=100", url, pageNo)
+		resp, err := f.client.Get(pageURL)
+		if err != nil {
```

Comment:
you can inline that check

Reply:
but I am using `resp` in many places. If I inline it this variable will have a scope of the if.